src/aggregation/agg_req_with_accessor.rs

```diff
@@ -244,6 +283,39 @@ impl AggregationWithAccessor {
                 )?;
                 add_agg_with_accessor(accessor, column_type, &mut res)?;
             }
+            TopHits(top_hits) => {
+                let (agg, top_hits) = {
+                    let top_hits =
+                        top_hits.validate_and_resolve(reader.fast_fields().columnar())?;
+                    (
+                        Aggregation {
```

Comment:
Why is top hits special and does need this overload mechanism

Reply:
Because of `docvalue_fields`, which to be fair should be moved to a common module and made available to other Top collectors too.

Reply:
you can clone `agg` at the root and make it mutable instead

Reply:
Done